### DISCUSSION TITLE
Why a spectral distribution read from a CSV file cannot be reshaped?

### DISCUSSION THREAD

**Discussion by frankkjensen** [colour-science/colour#1170] [Q&A] [answered]
Hi,

I load a spectra with:
`sds = colour.read_sds_from_csv_file('Cyan.csv', delimiter='\t', header=None, names=['WL', 'Cyan'])`
And then I try to reshape it with:
`colour.colorimetry.reshape_sd(sds['Cyan'], colour.SpectralShape(360, 830, 1))`
But I get this error: 
`AssertionError: "start" attribute value must be strictly less than "780"!`

What am I doing wrong?

I have attached the csv file:
[Cyan.csv](https://github.com/colour-science/colour/files/11798092/Cyan.csv)

-Frank

**Comment by thomasmansencal** [comment]
Hi @frankkjensen,

I think it is an issue with the current code that now uses Numpy: If you look at the spectral distribution, the first wavelength is `nan`:

```
[[             nan   0.00000000e+00]
 [  3.58338471e+02   2.60061000e-04]
 [  3.58638672e+02   9.62139000e-04]
 ..., 
 [  8.31245300e+02   1.45844600e-03]
 [  8.31516602e+02   1.42960800e-03]
 [  8.31787903e+02   9.48240000e-04]]
```

Try to set the header directly and remove your two calls that set it for the time being:

```csv
WL	Cyan
358.038269	0
358.3384705	0.000260061
358.6386719	0.000962139
```

I will try to understand what is going on!

**Reply by thomasmansencal** [reply]
Confirming that `np.genfromtxt` is the issue:

```python
import numpy as np
print(np.genfromtxt('/Users/kelsolaar/Downloads/Cyan.csv', delimiter='\t'))
```

**Reply by thomasmansencal** [reply, marked as answer]
I found out! You have 3 bytes that cannot be decoded at the start of your file, so the first wavelength ends up as `nan`:

![image](https://github.com/colour-science/colour/assets/99779/4fa6c5b9-af91-41cd-a3b4-ba7631b04a7d)

**Reply by frankkjensen** [reply]
Thanks for looking into this.
The file is from SpecWin (spectrometer software from Instrument Systems). I looked up the "EF BB BF", and it should be the "byte order mark" for UTF-8 encoding, but maybe np.genfromtxt does not support that.
Anyway, as it was only 6 files, I did as you suggested and added the header in the files.

Thank for the support :-)
-Frank
